### PR TITLE
Configurable Gauge

### PR DESCRIPTION
**Pull request by objerke**
This pull request adds a new mod the the catalog folder. It is a gauge that can be configured in a number of ways:

- Arc width.
- Min and Max values via continuous axes.
- Start and end angle.
- Opacity of background arc.
- Opacity of scale ticks.
- Show gauge value as percent
- Shake overfull gauge.
- Show min and max labels.

A demo of the mod can be seen [here](https://spotfire-next.cloud.tibco.com/spotfire/wp/OpenAnalysis?file=/Users/2mjfcsoowfkhqwy6tida5fspqmesl5ht/Public/new-gauge&configurationBlock=SetPage%28pageIndex%3D0%29%3B&options=13-1,10-1,9-1,5-1,6-0,17-1,11-1,12-1,14-1,1-0,3-0,18-0,7-1,15-1,19-1,4-1,2-1). 
gauges-demo

gauges-demo

### CODE REVIEW

File: catalog/configurable-gauge/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Gauge mod
+
+This is a mod example demonstrating a gauge written in [`TypeScript`](https://www.typescriptlang.org/), rendered with [`d3`](https://d3js.org/) and bundled with [`esbuild`](https://esbuild.github.io/).
```
Comment (ebrandin) [inline catalog/configurable-gauge/README.md:3]: A catalog Mod should perhaps not be regarded an "example demonstrating".
"This is a gauge mod written in..."

Comment (hski-github) [reply catalog/configurable-gauge/README.md:3]: Btw, same „example“  language in all the other catalog mods READMEs.

File: catalog/configurable-gauge/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- In Spotfire, follow the steps of creating a new mod and then browse for, and point to, the _manifest_ in the `dist` folder.
+
+## Bundle for production
+
+The default build task will create an uncompressed JavaScript bundle to simplify development and debugging. When the mod is ready to be saved into the analysis file the JavaScript bundle should be compressed. By invoking `npm run build`, `esbuild` will create a minified bundle.
+
+## Unit tests
+
+Invoking `npm test` will start a test runner with a file watcher.
+
```
Comment (ebrandin) [inline catalog/configurable-gauge/README.md:56]: Nice that there are unit tests. Unfortunately 30 of them fails.
Test "should render the arc within the gauge group with" fails for pad angles 0, 60 and 90.

File: catalog/configurable-gauge/src/settings.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                        type="range"
+                        max={c.max}
+                        min={c.min}
+                        value={c.property.value<number>()!}
+                        step={c.step}
+                        onChange={function (this: HTMLInputElement, e: any) {
+                            const { value } = e.target;
+                            let p = parseInt(value);
+                            console.log(this, e, c, p);
+                            c.property.set(p);
```
Comment (ebrandin) [inline catalog/configurable-gauge/src/settings.tsx:79]: Remove console.log

File: catalog/configurable-gauge/static/mod-manifest.json [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            "defaultValue": 100,
+            "type": "integer"
+        },
+        {
+            "name": "showPercent",
+            "defaultValue": "false",
+            "type": "boolean"
+        },
+        {
+            "name": "showShake",
```
Comment (ebrandin) [inline catalog/configurable-gauge/static/mod-manifest.json:34]: Consider renaming to "vibrate", or similar. Both here and in the UI. I think that would be easier to understand.

File: catalog/configurable-gauge/tests/gauge.test.ts [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    formattedValue: "10 000",
+    key: "second",
+    label: "Second",
+    mark() {},
+    maxLabel: "20 000",
+    minLabel: "0",
+    percent: 0.5
+};
+
+let padAngles = [0, 5, 10, 15, 20, 30, 35, 40, 45, 50, 60, 70, 80, 90];
```
Comment (ebrandin) [inline catalog/configurable-gauge/tests/gauge.test.ts:41]: padAngled 0, 60, 90 fails in unit test. Don't know why yet.

**Pull request merged**